support/azure/azure-kubernetes/availability-performance/node-auto-repair-errors.md

```diff
@@ -0,0 +1,39 @@
+ms.custom: sap:Node/node pool availability and performance
+---
+# Troubleshoot common node auto-repair errors
+
+When AKS detects a node which reports the NotReady status for more than 5 minutes, we will attempt to automatically repair your node. Node auto-repair is a best-effort service and does not guarantee that your node will be restored to a healthy state. Learn more about the node auto-repair process [here](https://learn.microsoft.com/en-us/azure/aks/node-auto-repair).
+
+During this process, AKS will initiate reboot, reimage, and redeploy actions on your unhealthy node. These repair actions may encounter errors due underlying causes, which will result in an error code. This article will discuss common errors with their potential causes and next steps, as well as best practices for monitoring node auto-repair.
+
+## Prerequisites
+To determine what type of node auto-repair error has occured, you will need to look for the following Kubernetes event: 
+"Node auto-repair [reboot/reimage/redeploy] action failed due to an operation failure: [error code]."
+
+> [!NOTE]
+> In most cases, node auto-repair errors will not cause any impact to your cluster or applications. When encountering a node auto-repair error, we recommend that you attempt to repair the node yourself by following the instructions here: [Basic troubleshooting of Node Not Ready failures](https://learn.microsoft.com/en-us/troubleshoot/azure/azure-kubernetes/availability-performance/node-not-ready-basic-troubleshooting). If you are unable to restore your node back to a Succeeded status and are noticing persistent errors reported by node auto-repair, you may be encountering a rare issue and should reach out to Azure support.
+
+## Common error codes
+The table below contains the most common node auto-repair errors.
+
+| Error code | Causes & Solution |
+|---|---|
+| ARM ErrorCode: VMExtensionProvisioningError | One or more VM extensions failed to be provisioned on the VM. Read more on possible error types and troubleshooting steps at [Troubleshoot the ERR_VHD_FILE_NOT_FOUND error code (124)](https://learn.microsoft.com/en-us/troubleshoot/azure/azure-kubernetes/create-upgrade-delete/error-code-vhdfilenotfound). This may also occur as a result of custom script extension (CSE) errors on the node. Learn more at [Troubleshoot node not ready failures caused by CSE errors](https://learn.microsoft.com/en-us/troubleshoot/azure/azure-kubernetes/availability-performance/node-not-ready-custom-script-extension-errors). |
+| ARM ErrorCode: InvalidParameter | This occurs due to a parameter causes errors when new nodes are created for an AKS cluster. Read more on causes and solution at [Troubleshoot the InvalidParameter error code](https://learn.microsoft.com/en-us/troubleshoot/azure/azure-kubernetes/create-upgrade-delete/error-code-invalidparameter#cause). |
```

Comment:
This might be confusing since customers did not make the request.

Generally, the remediator service requested a VM that no longer exists.

Reply:
Sounds good. Is there anything else we can say here about the customer impact - is this typically due to quota or capacity issues? And will remediator request a different VM or further redeployments will just fail?

Reply:
I've mentioned elsewhere, it can be spot VMs which get preempted frequently.
Otherwise it might have been deleted by cluster-autoscaler or something.